applications/GeoMechanicsApplication/custom_constitutive/linear_elastic_plane_strain_2D_law.cpp

```diff
@@ -89,12 +101,84 @@ void GeoLinearElasticPlaneStrain2DLaw::CalculatePK2Stress(const Vector& rStrainV
 {
     KRATOS_TRY
 
+    UpdateInternalDeltaStrainVector(rValues);
+
     Matrix C;
     this->CalculateElasticMatrix(C, rValues);
-    // Total formulation
-    noalias(rStressVector) = prod(C, rStrainVector);
+
+    // Incremental formulation
+    noalias(mStressVector) = mStressVectorFinalized + prod(C, mDeltaStrainVector);
```

Comment:
Maybe we can have a short discussion on if we need the two separate members (normal + Finalized). To me this makes it a bit more confusing what is really happening.

Reply:
A Newton Raphson like process iterates in the way that every iteration tries something that starts from the beginning of the step ( the Finalized values here ) tries an increment ( e.g. mDeltaStrainVector ) and produces a new value for the end of the step ( e.g. mStressVector ). After acceptance of the step result ( i.e. the step has converged) the finalized values are overwritten by the predicted end of step values.